packages/components/tooltip/src/useTooltip.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+
+import {
+  useFloating,
+  autoUpdate,
+  offset,
+  useHover,
+  useFocus,
+  useDismiss,
+  useRole,
+  useInteractions,
+  autoPlacement,
+  arrow,
+  shift,
+  flip,
+  type Placement,
+} from '@floating-ui/react';
+import type { TooltipPlacement } from './Tooltip';
+
+export interface TooltipOptions {
+  placement?: TooltipPlacement;
+  showDelay?: number;
+  hideDelay?: number;
+  usePortal?: boolean;
+  isVisible?: boolean;
+}
+
+export function useTooltip({
+  isVisible = false,
+  placement = 'auto',
+  showDelay,
+  hideDelay,
+  usePortal,
+}: TooltipOptions = {}) {
+  const arrowRef = React.useRef(null);
+  const [isOpen, setIsOpen] = React.useState(isVisible);
+
+  let sanitizedPlacement: Placement = 'top';
+  const middleware = [];
```

Comment:
This should be wrapped in `useMemo` to prevent constant rerendering due to the newly created array.